### PR TITLE
get content item by external id

### PR DESCRIPTION
**Pull request by AndroidDoctorr**

### CODE REVIEW

File: content-review/api/src/main/java/org/sakaiproject/contentreview/service/ContentReviewQueueService.java [hunk 1]
```diff
@@ -104,6 +104,14 @@ int getReviewScore(Integer providerId, String contentId)
 	 * @return {@code Optional<ContentReviewItem>}
 	 */
 	Optional<ContentReviewItem> getQueuedItem(Integer providerId, String contentId);
+	
+	/**
+	 * Get an item that has been queued and contentId.
```
Comment (baholladay) [inline content-review/api/src/main/java/org/sakaiproject/contentreview/service/ContentReviewQueueService.java]: This description doesn't make sense

File: content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java [hunk 2]
```diff
@@ -77,6 +77,16 @@
...
 	}
+	
+	public Optional<ContentReviewItem> findByProviderAndExternalId(Integer providerId, String externalId) {
+
+		Criteria c = sessionFactory.getCurrentSession()
+				.createCriteria(ContentReviewItem.class)
+				.add(Restrictions.eq("providerId", providerId))
+				.add(Restrictions.eq("externalId", externalId));
+		
+		return Optional.ofNullable((ContentReviewItem) c.uniqueResult());
```
Comment (baholladay) [inline content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java:88]: did you test this function to see if it works?

File: content-review/impl/service/src/main/java/org/sakaiproject/contentreview/service/ContentReviewQueueServiceImpl.java [hunk 3]
```diff
@@ -189,6 +189,18 @@ public void removeFromQueue(Integer providerId, String contentId) {
...
 	}
+	
+	/* (non-Javadoc)
+	 * @see org.sakaiproject.contentreview.service.ContentReviewQueueService#getQueuedItem(java.lang.Integer, java.lang.String)
+	 */
+	@Override
+	@Transactional(readOnly=true)
+	public Optional<ContentReviewItem> getQueuedItemByExternalId(Integer providerId, String externalId) {
+		Objects.requireNonNull(providerId, "providerId cannot be null");
+		Objects.requireNonNull(externalId, "contentId cannot be null");
```
Comment (baholladay) [inline content-review/impl/service/src/main/java/org/sakaiproject/contentreview/service/ContentReviewQueueServiceImpl.java]: warning message doesn't match

**Comment by baholladay** [comment]
code looks good, make the few changes I mentioned. Also, let's add an index for ProviderId + ExternalId... in large schools, this will be needed. You can do this in ./api/src/main/resources/ContentReviewItem.hbm.xml and will need a migration path as well.

**Comment by baholladay** [comment]
Let's defer the index work in ticket https://turnitin.atlassian.net/browse/INT-11180 since Ben will need this PR today

File: content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java (hunk 2 above)
Comment (AndroidDoctorr) [reply content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java:88]: yep

**Pull request merged**